Add stats granularity spinner to Traffic tab

irfano opened the pull request:
This adds a stats granularity spinner to the Traffic tab and removes the timezone from the date selector bar.
This only adds the view, itemSelectedListener will be implemented in another PR.

**Note for the reviewer:**
The dates on the date selector bar are aligned to the left when the `stats_traffic_tab` flag is disabled. To align dates to the left when the flag is disabled and to the right when the flag is enabled, `stats_date_selector.xml` has been updated, and `Space` views have been added to the layout.

<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/2471769/ccaf900a-6e15-44bc-8680-972e195f07f7" width=300>

The same spinner view from the People screen has been used on the stats screen.

<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/2471769/d09c150e-f519-4492-8f36-d7ca5b5557fc" width=300>

-----

## To Test:
1. Log in.
2. Navigate to "My Site → Stats → DAYS"
5. Verify the date selector bar does not include granularity spinner.
6. Navigate to "Me → Debug settings"
7. Check `stats_traffic_tab` on REMOTE FEATURES tab.
8. Navigate to "My Site → Stats → DAYS"
9. Verify the date selector bar has a granularity spinner and the layout looks as expected.

<!-- Test instructions per dependency update: https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/docs/test_instructions_per_dependency_update.md -->

-----

## Regression Notes

1. Potential unintended areas of impact

    - Date selector bar when the `stats_traffic_tab` flag is disabled.

6. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Tested when the `stats_traffic_tab` flag is disabled.

8. What automated tests I added (or what prevented me from doing so)

    - The addition of a new spinner is not worth including a new UI test.

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## UI Changes Testing Checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

irfano committed:
Add spinner labels for stats granularity

irfano committed:
Update `stats_date_selector.xml` by adding spinner

This layout has been also updated to support gone spinner state.

irfano committed:
Initialize stats granularity spinner from the fragment

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/275">24.1</a>. This milestone is due in less than 4 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ravishanker reviewed (approved):
LGTM 👍 

Are you going to add background around Spinner, and also hid currentTimeZone label or in next PR?

irfano commented:
👋🏻 @ravishanker, I don't think we should add a background to the spinner. The Figma design was made for iOS, and our existing spinner design doesn't have a background; see the one on the People screen. Wdyt, should we add a background?

> also hid currentTimeZone label or in next PR?

I kept the timezone intentionally. There's a logic for displaying the timezone label; it's gone if the site response doesn't include `timezone`. 

However, after checking the stats screen on the web, I noticed there isn't a timezone displayed there. I believe we can safely remove it. @guarani, have you removed the datepicker on iOS? If not, do you think we should remove it?

guarani commented:
We decided to leave the timezone ~picker~ label out after discussing in p1705672397461839/1705670832.178909-slack-C06BR07TJHK.

irfano committed:
Hide timezone when stats_traffic_tab enabled

irfano committed:
Fix `StatsDateSelectorTest` by adding `stats_traffic_tab` flag

irfano committed:
Add `isGranularitySpinnerVisible` parameter to `DateSelectorUiModel`

irfano committed:
Move logic of granularity spinner visibility to `DateSelectorViewUtils`

irfano commented:
- @ravishanker, I have hidden the timezone everywhere when the flag is enabled: 3945f06b41d25a772740f909173bb541d44cae0
- There was a bug when you approved the PR. The granularity spinner was visible on other screens, such as the "This year"s card detail, even when the feature flag was disabled. I moved the visibility logic to `DateSelectorViewUtils`, and it fixed the bug. c9a48b1afc852bc2ec5197cf4868ab1883edf00d

Could you please re-review te PR?

ravishanker commented:
> * @ravishanker, I have hidden the timezone everywhere when the flag is enabled: [3945f06](https://github.com/wordpress-mobile/WordPress-Android/commit/3945f06b41d25a772740f909173bb541d44cae06)
> * There was a bug when you approved the PR. The granularity spinner was visible on other screens, such as the "This year"s card detail, even when the feature flag was disabled. I moved the visibility logic to `DateSelectorViewUtils`, and it fixed the bug. [c9a48b1](https://github.com/wordpress-mobile/WordPress-Android/commit/c9a48b1afc852bc2ec5197cf4868ab1883edf00d)
> 
> Could you please re-review te PR?

👋 @irfano - The spinner has completely disappeared, not showing at all after this change!

irfano committed:
Add `TrafficListViewModel` to fix the visibility logic of spinner

irfano committed:
Fix initializing view model for Annual stats section

irfano commented:
Good catch, @ravishanker! I quickly added `TrafficListViewModel` to address the visibility issue with the spinner. However, this change breaks the date selector on the Traffic tab, preventing the selection of previous dates. I will handle it when adding functionality to change the granularity of the tab with spinner by implementing `TrafficListViewModel` properly.

Additionally, I noticed that 0f580c19085d402c0ec4c15616308832c546ec7b commit that we introduced this sprint changed `StatsSection.ANNUAL_STATS`'s ViewModel from `InsightsListViewModel` to `DayListViewModel`. I believe this was not your intention in the commit. It doesn't affect the annual stats card since it uses `StatsViewAllFragment` and `StatsViewAllViewModel` but I still I reverted that commit: 0886b49a45d6acaf13177f8d2762c63474ad3908

ravishanker reviewed (approved):
Spinner is working.  LGTM 👍
